go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -271,12 +274,31 @@ func (qre *QueryExecutor) checkPermissions() error {
 	}
 
 	// empty table name, do not need a table ACL check.
-	if qre.plan.TableName().IsEmpty() {
+	if qre.plan.TableName().IsEmpty() && qre.plan.NewName != nil && qre.plan.NewName.IsEmpty() {
 		return nil
 	}
-
+	if qre.plan.NewName != nil && !qre.plan.NewName.IsEmpty() {
+		altAuthorized := tableacl.Authorized(qre.plan.NewName.String(), qre.plan.PlanID.MinRole())
+		altTableACLStatsKey := []string{
+			qre.plan.NewName.String(),
+			altAuthorized.GroupName,
+			qre.plan.PlanID.String(),
+			callerID.Username,
+		}
+		if !altAuthorized.IsMember(callerID) {
+			if qre.tsv.qe.enableTableACLDryRun {
+				tabletenv.TableaclPseudoDenied.Add(altTableACLStatsKey, 1)
+			} else if qre.tsv.qe.strictTableACL {
+				errStr := fmt.Sprintf("table acl error: %q cannot run %v on table %q", callerID.Username, qre.plan.PlanID, qre.plan.NewName)
+				tabletenv.TableaclDenied.Add(altTableACLStatsKey, 1)
+				qre.tsv.qe.accessCheckerLogger.Infof("%s", errStr)
+				return vterrors.Errorf(vtrpcpb.Code_PERMISSION_DENIED, "%s", errStr)
+			}
+			// possibly record somehow that it passed the altAuthorized check
+		}
+	}
 	if qre.plan.Authorized == nil {
-		return vterrors.Errorf(vtrpcpb.Code_PERMISSION_DENIED, "table acl error: nil acl")
+		return vterrors.Errorf(vtrpcpb.Code_PERMISSION_DENIED, "table acl error: nil acl") // should be panic. nil can't happen
```

Comment:
nit: if we're going to keep this like this and want to comment it, i'd prefer the comment go above the line. Otherwise it creates a long line that wraps on split review view